tests/test_fetcher.py

```diff
@@ -183,4 +193,23 @@ def test_github_fetcher(url, username, repository, git_ref, spec, tmp_path):
 @pytest.mark.xfail(raises=ValueError, strict=True)
 def test_invalid_github_fetcher(url, tmp_path):
     """Test handling of invalid GitHub URLs."""
-    _get_github_fetcher(url, tmp_path)
+    _get_github_fetcher(ParsedUrl(url), tmp_path)
+
+
+@pytest.mark.parametrize(
+    "url, expected_name",
+    [
+        (GIT_URL, "reana-demo-root6-roofit"),
+        (GIT_URL + "/", "reana-demo-root6-roofit"),
+        (GITHUB_REPO_URL, "reana-demo-root6-roofit"),
+        (GITHUB_REPO_URL + "/", "reana-demo-root6-roofit"),
+        (GITHUB_REPO_URL + "/tree/branch", "reana-demo-root6-roofit-branch"),
+        (GITHUB_REPO_URL + "/tree/branch/", "reana-demo-root6-roofit-branch"),
+        (GITHUB_REPO_ZIP, "master"),
```

Comment:
I also think it would be nice to have the repository's name in this case, but it is probably better if we tackle this issue when we will refactor the fetcher (see [this comment](https://github.com/reanahub/reana-server/pull/448#discussion_r822530150)). What do you think?

Reply:
Created issue: https://github.com/reanahub/reana-server/issues/453